src/SonarScanner.MSBuild.Common/Telemetry/TelemetryUtils.cs

```diff
@@ -33,6 +33,23 @@ public static void AddTelemetry(ILogger logger, AggregatePropertiesProvider aggr
         }
     }
 
+    public static void AddTelemetry(ILogger logger, HostInfo serverInfo)
+    {
+        if (serverInfo is null)
+        {
+            return;
+        }
+        var serverUrl = serverInfo.IsSonarCloud
+                        ? serverInfo.ServerUrl
```

Comment:
Even for sonar cloud, the url could be a company internal proxy server, that forwards to sonarcloud.io. You can use CloudHostInfo.IsKnownUrl, to make sure that this isn't a user proxy.